### PR TITLE
Jetpack Social: Add toggle to Social admin page to enable or disable SIG and pick a default template

### PR DESCRIPTION
**Pull request by danielpost**
## Proposed changes:

This PR adds a toggle to the Social admin page that can be used to enable or disable Social Image Generator. It also adds a button that opens a template picker for choosing a default template.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1202816441505085-as-1204116019981285/f
1202816441505085-as-1204097909752714/f

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Enable SIG on your site and open the Social admin page.
* Make sure that the toggle matches the screenshot.
* Check that the toggle updates the status by running `( new Automattic\Jetpack\Publicize\Social_Image_Generator\Settings() )->is_enabled();`
* * Check that the template picker updates the default template by running `( new Automattic\Jetpack\Publicize\Social_Image_Generator\Settings() )->get_default_template();`

image
image

### CODE REVIEW

**Review by gmjuhasz — CHANGES_REQUESTED** [review]
I think this is close. I do have another issue, if I open a post that was created before this change, on that post the template picker has nothing selected. Even if I select something, saving then reloading shows that none is selected. This also happens on new post, if I save them as draft and reload the page

CleanShot 2023-03-27 at 17 52 04 png

I am not sure yet what causes it

File: projects/plugins/social/src/class-jetpack-social.php [hunk 1]
```diff
@@ -318,7 +323,7 @@ class_exists( 'Jetpack' ) ||
 					),
 					'hasPaidPlan'                   => $publicize->has_paid_plan(),
 					'isEnhancedPublishingEnabled'   => $publicize->is_enhanced_publishing_enabled( Jetpack_Options::get_option( 'id' ) ),
-					'isSocialImageGeneratorEnabled' => $publicize->has_social_image_generator_feature(),
+					'isSocialImageGeneratorEnabled' => $publicize->has_social_image_generator_feature() && ( new Automattic\Jetpack\Publicize\Social_Image_Generator\Settings() )->is_enabled(),
```
Comment (pablinos) [inline projects/plugins/social/src/class-jetpack-social.php:326]: This is probably fine, but should we create a helper function that combines these two checks?

Also this removes the panel entirely, so we should probably do things like not register the meta when it's been disabled like this. Maybe we should have a `$publicize->is_social_image_generator_enabled` that can check the plan and the setting, like this, and we can use that where we're currently just checking for the feature.

**Review by pablinos — COMMENTED** [review]
This looks pretty good to me. The more I think about it, the more I feel we should default SIG to enabled. We could do that by adding `'enabled' => true` to [the default array](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/publicize/src/social-image-generator/class-settings.php#L44). I realise that means someone would have to install Social to turn it off, but currently they'd need to do that to turn it on. I'm not sure which way round is best!

Just one minor thought on how we check for SIG being enabled in the initial state.

**Review by gmjuhasz — APPROVED** [review]
For my previous comment:
We investigated that it was an issue with meta saving on trunk, not this pr.

We said that we will create a follow-up for the comments Paul mentioned above.

Other than that, looks good to me.

**Pull request merged**